crates/accelerate/src/sabre/route.rs

```diff
@@ -573,14 +600,16 @@ pub fn swap_map_trial(
         }
         if routable_nodes.is_empty() {
```

Comment:
We shouldn't necessarily, because the release valve should only be taken rarely, and we don't want to overallocate the happy path, and we should still only be routing two nodes in the new case.